src/uu/ptx/src/ptx.rs

```diff
@@ -95,12 +95,21 @@ impl WordFilter {
         if matches.opt_present("b") {
             crash!(1, "-b not implemented yet");
         }
-        let reg = if matches.opt_present("W") {
-            matches.opt_str("W").expect("parsing options failed!")
-        } else if config.gnu_ext {
-            "\\w+".to_owned()
+        // Ignore empty string regex from cmd-line-args
+        let arg_reg: Option<String> = if matches.opt_present("W") {
+            matches.opt_str("W").filter(|reg| !reg.is_empty())
         } else {
-            "[^ \t\n]+".to_owned()
+            None
+        };
+        let reg = match arg_reg {
+            Some(arg_reg) => arg_reg,
+            None => {
+                if config.gnu_ext {
+                    "\\w+".to_owned()
```

Comment:
Would it be possible to test this case too?
thanks

Reply:
easiest would be running uutils `ptx -G -R -W '[^ \t\n]+' input` (the default regex anyways) and have output be the same as the other tests - but then I realised this doesn't prove the cmd-line regex was really used.

So I changed the default regex to omit the space `[^\t\n]+` and used that as a test.

However before all this I tried to do `ptx -G -R -W '\w+' input` basically emulating GNU extensions "on" default regex - however the output of this does not match the output of GNU `ptx -G -R -W '\w+' input`, meaning there seems to be another bug, should I file this?